File: read_leads.py

import os
import psycopg2
from urllib.parse import urlparse

# Let's get DATABASE_URL from .env.local
with open(".env.local") as f:
    for line in f:
        if line.startswith("DATABASE_URL="):
            db_url = line.split("=", 1)[1].strip().strip('"')

conn = psycopg2.connect(db_url)
cursor = conn.cursor()
cursor.execute("""
    SELECT event_id, event_type, source, medium, campaign, city, gclid, wbraid, gbraid, event_timestamp
    FROM lead_events
    ORDER BY event_timestamp DESC
    LIMIT 20
""")
columns = [desc[0] for desc in cursor.description]
rows = cursor.fetchall()
import json
print(json.dumps([dict(zip(columns, r)) for r in rows], default=str, indent=2))
cursor.close()
conn.close()

File: read_leads.mjs

import dotenv from 'dotenv';
import pg from 'pg';
dotenv.config({ path: '.env.local' });

const { Client } = pg;
const client = new Client({
  connectionString: process.env.DATABASE_URL
});
await client.connect();

const res = await client.query('SELECT event_id, event_type, source, medium, campaign, city, gclid, wbraid, gbraid, event_timestamp FROM lead_events ORDER BY event_timestamp DESC LIMIT 20');
console.log(JSON.stringify(res.rows, null, 2));
await client.end();
